tools/builder/debian/Dockerfile

```diff
@@ -0,0 +1,155 @@
+# Stage 1: Create the base image with all necessary components
+FROM debian:9 as vagrant-base
+
+# Configure the debconf frontend.
+#
+# This image doesn't include whiptail, dialog, nor the readline perl module.
+# Therefore, the debconf frontend will be set to 'teletype' to avoid error
+# messages about no dialog frontend could be found.
+RUN echo 'debconf debconf/frontend select teletype' | debconf-set-selections
+
+# Install the necessary packages.
+#
+# In addition to the regular Debian base image, a BASIC set of packages from the
+# Debian minimal configuration will be installed. After all packages have been
+# installed, the apt caches and some log files will be removed to minimize the
+# image.
+#
+# NOTE: An upgrade will be performed to include updates and security fixes of
+#       installed packages that received updates in the Debian repository after
+#       the upstream image has been created.
+#
+# NOTE: No syslog daemon will be installed, as systemd's journald should fit
+#       most needs. Please file an issue if you think this should be changed.
+RUN apt-get update && \
+    apt-get dist-upgrade -y && \
+    apt-get install -y --no-install-recommends \
+        apt-transport-https \
+        ca-certificates \
+        sudo \
+        systemd      \
+        systemd-sysv \
+        cron         \
+        anacron \
+        curl \
+        wget \
+        build-essential \
+        gnupg \
+        openssh-server \
+        openssh-client && \
+    apt-get clean
```

Comment:
apt-get also non determinism, we should bake the needed packages into the base image

Reply:
This is the base image